src/powershell/Public/ConvertTo-FinOpsSchema/ConvertTo-FinOpsSchema.ps1

```diff
@@ -0,0 +1,221 @@
+
+.PARAMETER AmortizedCost
+Specifies the amortized cost data to be converted. Object must be a supported Microsoft Cost Management schema.
+
+.PARAMETER Version
+FOCUS schema version to convert data to. Allowed values: 0.5, 1.0-workingdraft, 1.0-preview. Default = 0.5.
+
+.EXAMPLE
+ConvertTo-FinOpsSchema -ActualCost (Import-Csv my-actual-cost-details.csv) -AmortizedCost (Import-Csv my-amortized-cost-details.csv) | Export-Csv my-cost-details-in-focus.csv
+
+.NOTES
+Version: 
+Author: 
+Date: 
+#>
+
+function ConvertTo-FinOpsSchema {
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory=$true)]
+        [object]$ActualCost,
+
+        [Parameter(Mandatory=$false)]
+        [object]$AmortizedCost,
+
+        [Parameter()]
+        [ValidateSet("0.5", "1.0-workingdraft", "1.0-preview")]
+        [string]$Version = "0.5"
+    )
+    
+    # Define the columns to add to the output
+    $columnsToAdd = @(
+        'ftk_AccountName', 
+        'ftk_AccountId', 
+        'ftk_AccountOwnerId', 
+        'ftk_AvailabilityZone', 
+        'ftk_EffectivePrice'
+        )
+    
+    # Combine ActualCost and AmortizedCost
+    $CombinedCost = $ActualCost + $AmortizedCost
+    Write-Host "Combined Cost Data Count: $($CombinedCost.Count)"
+    
+    # Perform transformations based on the version
+    switch ($Version) {
+        "0.5" {
+            if ($null -ne $CombinedCost -and $CombinedCost.Count -gt 0) {
+                $firstRow = $CombinedCost[0]
+                # If the first row of the data does not contain the billingAccountId or BillingAccountId column, throw an error
+                if (-not ($firstRow.PSObject.Properties.Name -contains 'billingAccountId' -or $firstRow.PSObject.Properties.Name -contains 'BillingAccountId')) {
+                    throw "Cost details schema not supported."
+                }
+            # else throw an error 
+            } else {
+                throw "Combined Cost data is empty or null."
+            }
+            # TODO: Filter the actual cost data to rows where ChargeType == "Purchase" and PricingModel == "Reservation" or "SavingsPlan" (leave amortized cost as-is).
+            # This to ensure that we only have rows with Purchase, reservation or savings plan data. Only applicable to actual cost data. Leaving the amortized cost data as-is.
+
+            # 1. Uppercase Column Names Modifications
+            $CombinedCost = $CombinedCost | ForEach-Object {
+                $obj = $_
+                $newObj = New-Object PSObject
+                foreach ($property in $obj.PSObject.Properties) {
+                    # If property name starts with a lowercase letter, modify the name
+                    $newPropertyName = if ($property.Name -cmatch "^[a-z]") {
+                        $property.Name -creplace '^[a-z]', { $_.Value.ToUpper() }
+                    } else {
+                        # If not, keep the original name
+                        $property.Name
+                    }
```

Comment:
Is it simpler to just use substring? Not sure if there's a perf impact of doing this when not needed, but regex also has it's own perf issue 🤔
```suggestion
                    $newPropertyName = $property.Name.Substring[0].ToUpper() + $property.Name.Substring(1)
```